### PR TITLE
lint multidoc yaml to improve kubernetes usability

### PR DESCRIPTION
**Pull request by dexhorthy**
- [x] If any changes have been made to `/project` files, the corresponding `/src` files have been regenerated.
- [x] If any changes need to be deployed, the version has been bumped in `package.json`.
- [x] If any changes impact the replicated-lint package, they have been added to `CHANGELOG`.

### CODE REVIEW

File: src/lint.ts [hunk 1]
```diff
@@ -246,26 +245,26 @@ export function hackLintMultidoc(inYaml: string, maybeOpts?: MultidocLintOpts):
...
-  return _.map(docs, (doc, index) => {
-    const linted = new Linter(doc, {
-      rules: opts.rules,
-      registry: opts.registry,
-      lineColumnFinder,
-      offset,
-    }).lint();
-
+  let index = 0;
+  for (const doc of docs) {
```
Comment (dexhorthy) [inline src/lint.ts:243]: wow this is wrong.

Comment (dexhorthy) [reply src/lint.ts:243]: nope, nevermind. its right.

File: src/lint.ts [hunk 2]
```diff
@@ -246,26 +245,26 @@ export function hackLintMultidoc(inYaml: string, maybeOpts?: MultidocLintOpts):
...
+    }
     offset += doc.length + DOC_SEPARATOR_LENGTH;
-    return ({
-      index,
-      findings: linted,
-    });
-  });
+    index += 1;
+  }
+  return [];
```
Comment (dexhorthy) [inline src/lint.ts]: gonna change to return a single error if the doc isn't found.

File: src/lint.ts [hunk 3]
```diff
@@ -246,26 +239,35 @@ export function hackLintMultidoc(inYaml: string, maybeOpts?: MultidocLintOpts):
...
-  return _.map(docs, (doc, index) => {
-    const linted = new Linter(doc, {
-      rules: opts.rules,
-      registry: opts.registry,
-      lineColumnFinder,
-      offset,
-    }).lint();
-
+  let index = 0;
+  for (const doc of docs) {
```
Comment (ebramanti) [inline src/lint.ts:243]: If you need to keep track of the index, I would say just use a traditional for-loop

**Review by ebramanti — COMMENTED** [review]
One suggestion, a lot of formatting stuff too so I might have missed something.

File: src/lint.ts (hunk 3 above)
Comment (dexhorthy) [reply src/lint.ts:243]: yeah I guess. I don't know where my irrational (?) dislike for traditional for loops comes from.

**Pull request merged**